File: college classes/lab1/bookId.cpp
#include <iostream>

using namespace std;

struct vals{
    char name[20];
    int id;
};

int main()
{
    struct vals lib[5];
    
    for(int i=0;i<5;i++){
        
        cout<<"Enter the name of the book: ";
        cin>>lib[i].name;
        cout<<"Enter book id: ";
        cin>>lib[i].id;
        
    }
    
    cout<<endl<<endl;
    
    for(int i=0;i<5;i++){
        
        cout<<"Book : "<<lib[i].name<<" Book id: "<<lib[i].id<<endl;
        
    }

    for(int i=0;i<5-1;i++){
        
        for(int j=0;j<5-i-1;j++){
            
            if(lib[j].id>lib[j+1].id){
                
                struct vals temp;
                
                temp = lib[j+1];
                
                lib[j+1] = lib[j];
                
                lib[j] = temp;
            }
            
        }
        
    }

    cout<<endl<<endl;
    
    for(int i=0;i<5;i++){
        
        cout<<"Book : "<<lib[i].name<<" Book id: "<<lib[i].id<<endl;
        
    }

    
    return 0;
}

File: college classes/lab1/bookid.java
import java.util.Scanner;
public class bookid{

    static class hash{
        int id;
        String name;
    }
    public static void main(String[] args) {
        System.out.println("Enter the no of books wants to enter");
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();

        hash[] obj=new hash[n];

        for(int i=0;i<n;i++){
            obj[i]=new hash();
            System.out.print("Enter the book id : ");
            obj[i].id=sc.nextInt();
            System.out.println();
            System.out.print("Enter the book name : ");
            obj[i].name=sc.next();
            System.out.println();
        }
        
        obj=bubbleSort(obj);

        display(obj);
    }

    static void display(hash[] obj){
        for(int i=0;i<obj.length;i++){
            System.out.println("Book id is : "+obj[i].id);
            System.out.println("Book name is : "+obj[i].name);
        }
    }

    static hash[] bubbleSort(hash[] obj){
        
        for(int i=0;i<obj.length-1;i++){
            for(int j=0;j<obj.length-i-1;j++){
                if(obj[j].id>obj[j+1].id){
                    hash temp=obj[j];
                    obj[j]=obj[j+1];
                    obj[j+1]=temp;
                }
            }
        }

        return obj;
    }
    
}
